Code Quality: Clean up Files.Actions phase 1

0x5bfa opened the pull request:
## Code Quality: Clean up Files.Actions

### Motivation & Context

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

ferrariofilippo commented on src/Files.App/Actions/Content/Archives/CompressIntoArchiveAction.cs:
```diff
@@ -1,25 +1,22 @@
+		private readonly IContentPageContext context;
 
-		public string Label => "CreateArchive".GetLocalizedResource();
+		public string Label
+			=> "CreateArchive".GetLocalizedResource();
```
I remember there was some kind of issue with the rendering using arrow instead of `{ get; }`. I'll let you know

ferrariofilippo commented on src/Files.App/Actions/Content/PlayAllAction.cs:
```diff
@@ -1,30 +1,26 @@
-		public string Label { get; } = "PlayAll".GetLocalizedResource();
+		public string Label
+			{ get; } = "PlayAll".GetLocalizedResource();
 
-		public string Description { get; } = "PlayAllDescription".GetLocalizedResource();
+		public string Description
+			{ get; } = "PlayAllDescription".GetLocalizedResource();
 
-		public RichGlyph Glyph { get; } = new("\uE768");
+		public RichGlyph Glyph
+			{ get; } = new("\uE768");
```
Shouldn't you replace `{ get; }` even here?

0x5bfa replied:
Fixed

ferrariofilippo commented on src/Files.App/Actions/FileSystem/OpenItemAction.cs:
```diff
@@ -70,7 +73,7 @@ public OpenItemWithApplicationPickerAction()
 
 		public async Task ExecuteAsync()
 		{
-			NavigationHelpers.OpenSelectedItems(context.ShellPage, true);
+			await NavigationHelpers.OpenSelectedItems(context.ShellPage, true);
 		}
```
You can remote `async` from the declaration and then return `NavigationsHelpers.Open...`

0x5bfa replied:
Done

ferrariofilippo commented on src/Files.App/Actions/FileSystem/OpenItemAction.cs:
```diff
@@ -40,7 +37,7 @@ public OpenItemAction()
 
 		public async Task ExecuteAsync()
 		{
-			NavigationHelpers.OpenSelectedItems(context.ShellPage);
+			await NavigationHelpers.OpenSelectedItems(context.ShellPage);
 		}
```
As above

0x5bfa replied:
Done

ferrariofilippo commented on src/Files.App/Actions/FileSystem/CutItemAction.cs:
```diff
@@ -31,11 +26,10 @@ public CutItemAction()
 		{
 			if (context.ShellPage is not null)
-				UIFilesystemHelpers.CutItem(context.ShellPage);
-			return Task.CompletedTask;
+				await UIFilesystemHelpers.CutItem(context.ShellPage);
```
You can remove `async` and then use something like:
```
    return context.ShellPage is not null
        ? UIFilesystemHelpers.CutItem(context.ShellPage)
        : Task.CompletedTask;
```

0x5bfa replied:
CutItem is async operation

0x5bfa replied:
Done

ferrariofilippo commented on src/Files.App/Actions/Display/GroupAction.cs:
```diff
@@ -1,137 +1,173 @@
 	{
-		protected IContentPageContext ContentContext { get; } = Ioc.Default.GetRequiredService<IContentPageContext>();
-		protected IDisplayPageContext DisplayContext { get; } = Ioc.Default.GetRequiredService<IDisplayPageContext>();
+		protected IContentPageContext ContentContext = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+		protected IDisplayPageContext DisplayContext = Ioc.Default.GetRequiredService<IDisplayPageContext>();
```
Can you move the assignment to the constructor?

0x5bfa replied:
Fixed

ferrariofilippo commented on src/Files.App/Actions/Display/GroupAction.cs:
```diff
@@ -155,84 +193,111 @@ private void ContentContext_PropertyChanged(object? sender, PropertyChangedEvent
 	{
-		protected IContentPageContext ContentContext { get; } = Ioc.Default.GetRequiredService<IContentPageContext>();
-		protected IDisplayPageContext DisplayContext { get; } = Ioc.Default.GetRequiredService<IDisplayPageContext>();
+		protected IContentPageContext ContentContext = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+		protected IDisplayPageContext DisplayContext = Ioc.Default.GetRequiredService<IDisplayPageContext>();
```
As above

0x5bfa replied:
Fixed

ferrariofilippo reviewed (commented):
You should consider opening a different PR to refactor the initialization of `context` in most actions.
There are some little things that might be improved, but apart from that looks good

ferrariofilippo commented on src/Files.App/Actions/Content/Archives/DecompressArchiveToChildFolderAction.cs:
```diff
@@ -1,24 +1,19 @@
-using Files.App.Extensions;
-using Files.App.Helpers;
-using System.ComponentModel;
-using System.IO;
```
This is needed

0x5bfa replied:
Those were grayed out.

ferrariofilippo replied:
My bad, the only one needed was `System.IO` (which you already added back). I selected too many rows

ferrariofilippo commented on src/Files.App/Actions/Content/PreviewPopup/LaunchPreviewPopupAction.cs:
```diff
@@ -3,28 +3,33 @@
 	{
-		private readonly IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+		private readonly IContentPageContext context;
+
```
You forgot to assign it in the constructor

0x5bfa replied:
Fixed

0x5bfa commented:
> You should consider opening a different PR to refactor the initialization of context in most actions.

How come should I? I think it’s fine so far. Probably I will split up phase 1 PR into multiple PRs, so reviewers can review quickly and easily.

ferrariofilippo commented:
> How come should I? I think it’s fine so far.

There are a lot of `Actions` which use 
`private readonly IContentApplicationContext context = Ioc.Default.GetRequired<IContentPageContext>();`

I think the assignment should be moved to the constructor

0x5bfa commented:
> I think the assignment should be moved to the constructor

All of them were moved.

0x5bfa commented:
Ready for final review.

ferrariofilippo reviewed (changes requested):
There might be errors with tasks not awaited

0x5bfa commented on src/Files.App/Actions/Global/OpenHelpAction.cs:
```diff
@@ -1,26 +1,25 @@
 			var url = new Uri(Constants.GitHub.DocumentationUrl);
-			await Launcher.LaunchUriAsync(url);
+			return await Launcher.LaunchUriAsync(url);
 		}
```
Done

0x5bfa commented on src/Files.App/Actions/Navigation/CloseOtherTabsCurrentAction.cs:
```diff
@@ -1,38 +1,33 @@
-		private bool isExecutable;
-		public bool IsExecutable => isExecutable;
+		public bool IsExecutable
+			=> GetIsExecutable();;
```
Done

0x5bfa commented on src/Files.App/Actions/Open/OpenSettingsAction.cs:
```diff
@@ -17,16 +9,19 @@ internal class OpenSettingsAction : BaseUIAction, IAction
 			var dialog = dialogService.GetDialog(viewModel);
-			await dialog.TryShowAsync();
+			return await dialog.TryShowAsync();
 		}
```
Done

0x5bfa commented on src/Files.App/Actions/Navigation/NewTabAction.cs:
```diff
@@ -1,24 +1,31 @@
 
-		public HotKey HotKey { get; } = new(Keys.T, KeyModifiers.Ctrl);
+		public Task ExecuteAsync()
+		{
+			mainPageViewModel.AddNewTabAsync();
 
-		public Task ExecuteAsync() => mainPageViewModel.AddNewTabAsync();
+			return Task.CompletedTask;
+		}
```
Done

0x5bfa commented on src/Files.App/Actions/FileSystem/CreateShortcutAction.cs:
```diff
@@ -16,13 +16,16 @@ public string Description
-		public override bool IsExecutable
-			=> context.HasSelection && context.CanCreateItem && UIHelpers.CanShowDialog;
+		public override bool IsExecutable =>
+			context.HasSelection &&
+			context.CanCreateItem && UIHelpers.CanShowDialog;
```
Done

0x5bfa commented on src/Files.App/Actions/Favorites/UnpinItemAction.cs:
```diff
@@ -67,9 +59,10 @@ bool IsPinned(ListedItem item)
 		}
+
 		private void UpdateIsExecutable()
 		{
-			SetProperty(ref isExecutable, GetIsExecutable(), nameof(IsExecutable));
+			OnPropertyChanged(nameof(IsExecutable));
 		}
```
Done

0x5bfa commented on src/Files.App/Actions/Favorites/PinItemAction.cs:
```diff
@@ -73,7 +65,7 @@ bool IsPinnable(ListedItem item)
 		private void UpdateIsExecutable()
 		{
-			SetProperty(ref isExecutable, GetIsExecutable(), nameof(IsExecutable));
+			OnPropertyChanged(nameof(IsExecutable));
```
Done

ferrariofilippo commented on src/Files.App/Actions/FileSystem/OpenItemAction.cs:
```diff
@@ -68,12 +70,16 @@ public RichGlyph Glyph
-			await NavigationHelpers.OpenSelectedItems(context.ShellPage, true);
+			NavigationHelpers.OpenSelectedItems(context.ShellPage, true);
+
+			return Task.CompletedTask;
```
```suggestion
			return NavigationHelpers.OpenSelectedItems(context.ShellPage, true);
```

ferrariofilippo reviewed (commented):
Some more style-related issues, but I don't think they are blockers

0x5bfa commented on src/Files.App/Actions/Content/Archives/CompressIntoZipAction.cs:
```diff
@@ -45,9 +43,10 @@ public Task ExecuteAsync()
+			return
+				context.PageType != ContentPageTypes.RecycleBin &&
+				context.PageType != ContentPageTypes.ZipFolder &&
+				context.PageType!= ContentPageTypes.None;
```
Fixed

0x5bfa commented on src/Files.App/Actions/FileSystem/AddItemAction.cs:
```diff
@@ -1,33 +1,36 @@
 
-        private readonly IDialogService dialogService = Ioc.Default.GetRequiredService<IDialogService>();
+		private readonly IDialogService dialogService;
 
         private readonly AddItemDialogViewModel viewModel = new();
 
-		public string Label { get; } = "BaseLayoutContextFlyoutNew/Label".GetLocalizedResource();
+		public string Label
```
Fixed

0x5bfa commented on src/Files.App/Actions/FileSystem/AddItemAction.cs:
```diff
@@ -1,33 +1,36 @@
+			=> new(Keys.N, KeyModifiers.CtrlShift);
 
-		public RichGlyph Glyph { get; } = new(opacityStyle: "ColorIconNew");
+		public RichGlyph Glyph
```
Fixed

ferrariofilippo reviewed (approved):
LGTM

0x5bfa commented:
@yaira2 Ready to merge?

yaira2 commented:
@0x5bfa this looks like a big change, I'd like to get a second review before merging.

0x5bfa commented:
Should I have split up into like 3 PRs? I can do that instead of fixing conflicts and waiting reviews.

yaira2 commented:
We can do them together

yaira2 commented:
@0x5bfa besides for formatting, were there any other changes?

0x5bfa commented:
Change overview

- Moved initializers to the constructors (i.e. dependency injection initializings)
- Improved formatting
- Updated return values of Execute(); You can see this changing in ferrariofilippo's reviews
